experimental/relativetime/src/provider.rs

```diff
@@ -0,0 +1,99 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+// Provider structs must be stable
+#![allow(clippy::exhaustive_structs, clippy::exhaustive_enums)]
+
+//! Data provider struct definitions for this ICU4X component.
+//!
+//! Read more about data providers: [`icu_provider`]
+
+use alloc::borrow::Cow;
+use icu_provider::{yoke, zerofrom};
+use zerovec::ZeroMap;
+
+/// Relative time patterns data struct.
+#[icu_provider::data_struct(
+    StandardRelativeTimeV1Marker = "relativetime/standard@1",
+    NarrowRelativeTimeV1Marker = "relativetime/narrow@1",
```

Comment:
Thought: I'm wondering whether we want to slice this smaller. For example, I know ECMA-402 delays the unit choice to the format function, but should we be making it an option in the constructor instead? And how about the direction (past/future)? We're going to get some pretty big data keys out of this, which begs the question about whether we should be slicing more granularly.

It's OK to make constructors that load multiple data keys for convenience, of course.

One possible axis that is already in ECMA-402 is "numeric". If the user requests numeric: "always", then they shouldn't need to load the non-numeric data. So, we could maybe move that data into a separate data key.

Reply:
The data keys are pretty big currently with  most of the fields being 'None'. 
I'm in favour of splitting it further into at least | "unit" x "width" | data keys. 
This would remove the need for having an array of "RelativeTimePattern".
Then we can focus on an efficient representation of "PluralRulesCategory" => "Singular substitution pattern" mapping.

Reply:
Would be great to do unit x width (8*3). What do you think about unit x width x numeric (8*3*2)?

Yes, let's brainstorm the best representation for PluralCategory => Pattern.

There are some advantages to expanding to separate fields instead of using ZeroMap. Note that `None` fields do not take up much space in Postcard.